src/theme/utils.ts

```diff
@@ -0,0 +1,43 @@
+import math from "polished/lib/math/math"
+import { mergeDeepRight, path, pathOr } from "ramda"
+
+export type ThemeAtom = string | number | { [key: string]: string | number | ThemeAtom }
```

Comment:
Why does this Theme need to opt between all these types?
Wouldn't just a (simple) Theme interface do fine?

Reply:
@VLegakis coz, there are could be sublevels. Thishow recursive typing implemented here

Reply:
Theme could be constructed on application level, with keys/values we have no idea about right now. So, we use a wider type, also setting some constraints.